src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.h

```diff
@@ -65,6 +65,8 @@ template <class ImplClass>
 class GenericThreadStackManagerImpl_OpenThread
 {
 public:
+    static constexpr uint16_t kAttachNetworkTimeoutSeconds = 20;
```

Comment:
Why is this the right value.  The reason this value was chosen should at least be documented.  And I would think it should somehow be related to the ConnectMaxTimeSeconds of the network commissioning cluster.

Reply:
+1: this has to match exactly with ConnectMaxTimeSeconds source of truth

Reply:
I would suggest this be the source of truth for the network commissioning cluster, since this class the more close to the openthread apis as we have the following layers for Thread stack:

Cluster -> Driver -> GenericThreadStackManagerImpl_OpenThread